ui/src/pages/backoffice/Document/DocumentDetails/components/DocumentRelations/components/RelationRemover/RelationRemover.js

```diff
@@ -12,17 +13,41 @@ export default class RelationRemover extends Component {
   handleOpen = () => this.setState({ modalOpen: true });
 
   handleDelete = () => {
-    const { deletePayload, refererPid } = this.props;
+    const { related, referer, relationPayloadType } = this.props;
+
     this.setState({ modalOpen: false });
-    this.props.deleteRelations(refererPid, [deletePayload]);
+    let deletePayload = {};
+    if (relationPayloadType === 'siblings') {
+      deletePayload = {
```

Comment:
I would try to avoid as much as possible to prepare a REST API `payload` in a component. I would rather pass all params to the function, and let the `api` js get the params and transform them to the needed payload.

Reply:
👍 very good point